### PR TITLE
feat: unundeprecate p- and q-adic methods

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

**Comment by thofma** [comment]
@benlorenz I have a feeling I have asked this before, but is there a reason that we do not run the downstream tests with `depwarn=error`?

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name = "Nemo"
 uuid = "2edaba10-b0f1-5616-af89-8c11ac63239a"
-version = "0.45.3"
+version = "0.45.2"
```
Comment (lgoettgens) [inline Project.toml:3]:
```suggestion
version = "0.45.3"
```
You don't want to revert this

**Comment by lgoettgens** [comment]
We don't run it with deperrors by default, since we had some issues in the past with external dependencies of use (like BinaryBuilder.jl) deprecating stuff and then all of our CI was red until we adapted and released all of our packages. In a post-1.0-era of our upstream packages, deprecations can be used in a normal release cycle as well. It's just in the pre-1.0-era that we can only deprecate stuff when making a minor release. But this is something that we can just look for ourselves I think.

**Comment by thofma** [comment]
OK, makes sense, thans.

**Comment by fingolfin** [comment]
"unundeprecate" 😂 maybe "re-deprecate" ? 😁

**Pull request closed without merging**